### PR TITLE
Migrate from setup.cfg to pyproject.toml

### PR DESCRIPTION
**Pull request by azmeuk**
I was having a look at #1238 and I realized that setuptools+setup.cfg is not really convenient for build callbacks (such as: catalog compilation).
I was thinking about using hatch instead of setuptools to build packages, like [we did recently in wtforms](https://github.com/wtforms/wtforms/pull/798), as it is lightweight and provides build callbacks, and setuptools is deprecated anyways (since it is built on distutils, that have [just been deleted in python 3.12](https://docs.python.org/3/whatsnew/3.12.html?highlight=distutils)).

Anyways, if we use hatch in the future (or flit, or poetry, or whatever), then pyproject.toml is required, and if we keep using setuptools, pyproject.toml won't harm :)

We can have the discussion about the build tool on #1238

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,82 +0,0 @@
...
-
-[options.entry_points]
-flask.commands =
-    generate_password_hash = ihatemoney.manage:password_hash
-    generate-config = ihatemoney.manage:generate_config
-
-console_scripts =
-    ihatemoney = ihatemoney.manage:cli
-
-paste.app_factory =
```
Comment (azmeuk) [inline setup.cfg:81]: Is [paste deployment](https://docs.pylonsproject.org/projects/pastedeploy/en/latest/) still used?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+]
+
+[project.urls]
+"Source Code" = "https://github.com/spiral-project/ihatemoney"
+"Main instance" = "https://ihatemoney.org/"
+"Mobile app" = "https://ihatemoney.org/mobile"
+"Documentation" = "https://ihatemoney.readthedocs.io/en/latest/"
+
+[project.scripts]
+ihatemoney = "ihatemoney.manage:cli"
```
Comment (azmeuk) [inline pyproject.toml:85]: This makes the `ihatemoney` command available.

File: ihatemoney/manage.py [hunk 3]
```diff
@@ -32,14 +32,14 @@ def runserver(ctx):
     ctx.forward(run)
 
 
-@click.command(name="generate_password_hash")
+@cli.command(name="generate_password_hash")
```
Comment (azmeuk) [inline ihatemoney/manage.py:35]: This was needed to provide `ihatemoney generate_password_hash`

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    "Sphinx>=7.0.1,<8",
+    "docutils==0.20.1",
+    "myst-parser>=2,<3",
+]
+
+[project.urls]
+"Source Code" = "https://github.com/spiral-project/ihatemoney"
+"Main instance" = "https://ihatemoney.org/"
+"Mobile app" = "https://ihatemoney.org/mobile"
+"Documentation" = "https://ihatemoney.readthedocs.io/en/latest/"
```
Comment (azmeuk) [inline pyproject.toml:82]: `[project.urls]` allows for more urls to be displayed on the PyPI page.

File: setup.cfg (hunk 1 above)
Comment (almet) [reply setup.cfg:81]: I've checked and it seems that all the documented deployment options and configuration files are relying on the wsgi module, so we probably don't need to maintain this.

**Comment by zorun** [comment]
Thanks for the work @azmeuk 

@almet Could you please let time for other people to review changes before merging them extra-fast? In that case, I would have like to:

- first add more tests to catch the silent password hash generation crash
- then merge #1242 fixing the crash
- then release a bugfix release 6.1.2
- then merge more impacting changes that have a risk to introduce regression. Migrating to pyproject.toml does have such a risk (small but anyway)

**Comment by almet** [comment]
@zorun I'm sorry I didn't let you more time to review this. I understand what you mean and why it could have been simpler to merge it later on.

I believe these scenarii will probably happen more often than before, as more people are working on the repository than usual. We might want to setup some rules for merging, releasing, etc.

I see different ways to solve this, I've open #1245 to discuss that.

**Pull request merged**